pkg/controller/dnszone/gcpactuator.go

```diff
@@ -0,0 +1,176 @@
+
+// NewGCPActuator creates a new GCPActuator object. A new GCPActuator is expected to be created for each controller sync.
+func NewGCPActuator(
+	logger log.FieldLogger,
+	secret *corev1.Secret,
+	dnsZone *hivev1.DNSZone,
+	gcpClientBuilder gcpClientBuilderType,
+) (*GCPActuator, error) {
+	gcpClient, err := gcpClientBuilder(secret)
+	if err != nil {
+		logger.WithError(err).Error("Error creating GCPClient")
+		return nil, err
+	}
+
+	gcpActuator := &GCPActuator{
+		logger:    logger,
+		gcpClient: gcpClient,
+		dnsZone:   dnsZone,
+	}
+
+	return gcpActuator, nil
+}
+
+// Ensure GCPActuator implements the Actuator interface. This will fail at compile time when false.
+var _ Actuator = &GCPActuator{}
+
+// Create implements the Create call of the actuator interface
+func (a *GCPActuator) Create() error {
+	logger := a.logger.WithField("zone", a.dnsZone.Spec.Zone)
+	logger.Info("Creating managed zone")
+
+	zone := a.dnsZone.Spec.Zone
+	managedZone, err := a.gcpClient.CreateManagedZone(
+		&dns.ManagedZone{
+			Name:        generateManagedZoneName(zone),
+			Description: managedByHiveDescription,
+			DnsName:     controllerutils.Dotted(zone),
+		},
+	)
+
+	if err != nil {
+		logger.WithError(err).Error("Error creating managed zone")
+		return err
+	}
+
+	logger.Debug("Managed zone successfully created")
+	a.managedZone = managedZone
+	return nil
+}
+
+// Delete implements the Delete call of the actuator interface
+func (a *GCPActuator) Delete() error {
+	if a.managedZone == nil {
+		return errors.New("managedZone is unpopulated")
+	}
+
+	logger := a.logger.WithField("zone", a.dnsZone.Spec.Zone).WithField("zoneName", a.managedZone.Name)
+	logger.Info("Deleting managedzone")
+	err := a.gcpClient.DeleteManagedZone(a.managedZone.Name)
+	if err != nil {
+		log.WithError(err).Error("Cannot delete managed zone")
+	}
+	return err
+}
+
+// Exists implements the Exists call of the actuator interface
+func (a *GCPActuator) Exists() (bool, error) {
+	return a.managedZone != nil, nil
+}
+
+// UpdateMetadata implements the UpdateMetadata call of the actuator interface
+func (a *GCPActuator) UpdateMetadata() error {
+	// Nothing to do here since GCP CloudDNS doesn't support tags.
+	return nil
+}
+
+// ModifyStatus implements the ModifyStatus call of the actuator interface
+func (a *GCPActuator) ModifyStatus() error {
+	if a.managedZone == nil {
+		return errors.New("managedZone is unpopulated")
+	}
+
+	a.dnsZone.Status.GCP = &hivev1.GCPDNSZoneStatus{
+		ZoneName: &a.managedZone.Name,
+	}
+
+	return nil
+}
+
+// GetNameServers implements the GetNameServers call of the actuator interface
+func (a *GCPActuator) GetNameServers() ([]string, error) {
+	if a.managedZone == nil {
+		return nil, errors.New("managedZone is unpopulated")
+	}
+
+	logger := a.logger.WithField("zone", a.dnsZone.Spec.Zone)
+	result := a.managedZone.NameServers
+	logger.WithField("nameservers", strings.Join(result, ",")).Debug("found managed zone name servers")
```

Comment:
I don't think that we need to generate a string for the nameservers field. If we use `result` directly as the value of the field, then the field will have all of the name servers included. This would (1) make the code simpler and (2) avoid an unnecessary string formatting when not running the logger at debug level.